server/src/server.ts

```diff
@@ -97,24 +108,27 @@ interface ISettings {
     yseopml: IServerSettings;
 }
 
-// These are the example settings we defined in the client's package.json
-// file
+// These are the example settings we defined in the client's package.json file
 interface IServerSettings {
-    maxNumberOfProblems: number;
     activateParsingProblemsReporting: boolean;
     pathToPredefinedObjectsXml: string;
+    diagnosticLevel: string;
 }
 
 let engineModel: EngineModel;
-// hold the maxNumberOfProblems setting
 let pathToPredefinedObjectsXml: string;
 let activateParsingProblemsReporting: boolean;
-// The settings have changed. Is send on server activation
-// as well.
+
+// By default, the severity is Information.
+export let severity: DiagnosticSeverity = DiagnosticSeverity.Information;
+
+// The settings have changed. Is send on server activation as well.
 connection.onDidChangeConfiguration((change) => {
     const settings = change.settings as ISettings;
     activateParsingProblemsReporting = settings.yseopml.activateParsingProblemsReporting;
     pathToPredefinedObjectsXml = settings.yseopml.pathToPredefinedObjectsXml;
+    // One of the diagnostic level possible, or Information.
```

Comment:
You’ll have a bit of search-and-replace to do, I’m afraid. And the option’s name itself to fix.

> Diagnostic:
>
> * _adjective_ Of, relating to, or used in a diagnosis.
> * _adjective_ Serving to identify a particular disease; characteristic.
> * _noun_ The art or practice of medical diagnosis.
> * _noun_ A symptom or a distinguishing feature serving as supporting evidence in a diagnosis.
> * _noun_ An instrument or technique used in medical diagnosis.

# Edit

Wut, `DiagnosticSeverity` is not from you? Gosh. Then I dunno. Still, the option’s name is rather vague: it’s not the diagnosis of _everything_ but of parsing errors and the like, right? As soon as we’ll have something entirely different _X_ to diagnose, the option’s name will become confusing if the said option does not affect the level for _X_. But maybe I misunderstand that stuff.

Reply:
I'll rename the variable to `parsingIssueSeverityLevel`, something like that.